cyences_app_for_splunk/bin/rh_cyences_product_conf.py

```diff
@@ -80,22 +81,38 @@ def configure_saved_searches(self, enabled_products, disabled_products):
         savedsearches = self.get_saved_searches()
 
         changed_savedsearches = []
+        messages_to_display = []
+        errors = []
         for name, content in savedsearches.items():
             products = cs_utils.convert_to_set(content["action.cyences_notable_event_action.products"])
             current_disabled = cs_utils.is_true(content["disabled"])
 
-            if len(products&disabled_products)>0:
+            if len(products - disabled_products) == 0:
                 new_disabled = True
-            elif len(products&enabled_products)>0:
+            elif len(products & enabled_products) > 0:
                 new_disabled = False
             else:
                 new_disabled = current_disabled
-            
-            if current_disabled != new_disabled:
-                self.conf_manager.update_savedsearch(name, {"disabled": new_disabled})
-                changed_savedsearches.append(name + " => " + ("Disabled" if new_disabled else "Enabled"))
 
-        return '\n'.join(changed_savedsearches)
+            if current_disabled != new_disabled:
+                changed_savedsearches.append([name, new_disabled])
+                messages_to_display.append(name + " => " + ("Disabled" if new_disabled else "Enabled"))
+
+        if len(changed_savedsearches) > 0:
+            with concurrent.futures.ThreadPoolExecutor(max_workers=min(len(changed_savedsearches), 15)) as executor:
```

Comment:
Isn't 15 is high number? Shall we reduce it?

Reply:
It always gives error with 20 threads. I tested with 15 and it's working fine.